enterprise/server/scheduling/task_router/task_router.go

```diff
@@ -124,6 +176,10 @@ func (tr *taskRouter) RankNodes(ctx context.Context, action *repb.Action, cmd *r
 		return nonePreferred(nodes)
 	}
 
+	// Resample nodes by CPU weight so that nodes with more resources
+	// are more likely to get more tasks.
+	nodes = weightedResample(nodes)
```

Comment:
I think in the `nonePreferred` cases above, we still want to do a `weightedResample`, no? why not do the weightedResample on line 108 and keep the existing `nonePreferred` code that directly returns the nodes?